### PR TITLE
build with gcc-11 and clang-14

### PR DESCRIPTION
**Pull request by pheiduck**
update ubuntu to 22.04 and clang to 14

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -112,7 +112,7 @@ jobs:
 
   build-ubuntu-latest-pcp:
     # Turns out 'ubuntu-latest' can be older than 20.04, we want PCP v5+
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (BenBE) [inline .github/workflows/ci.yml:115]: Please note the comment there. Maybe update to `ubuntu-22.04` instead?

**Review by BenBE — COMMENTED** [review]
Any new static analyzer improvements / findings this triggers?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pheiduck) [reply .github/workflows/ci.yml:115]: `ubuntu-22.04` doesn't exist as Github virtual environment unitl now, maybe on August.

Comment (pheiduck) [reply .github/workflows/ci.yml]: Done: https://github.com/htop-dev/htop/pull/981/commits/79a329d1602fd47fd5431c2b2bd15f3d98d0e2c2

**Comment by BenBE** [comment]
I know this is somewhat nitpicking here, but could you please update your two commits, so the first one doesn't touch the ubuntu CI image selector? Commits should be specific to one thing they fix and leave other aspects untouched. TIA.

**Comment by BenBE** [comment]
Public Beta for Ubuntu 22.04 images started:
https://github.com/actions/virtual-environments/issues/5490

@pheiduck : The cleanup for the patch series is still pending.

**Comment by pheiduck** [comment]
@BenBE this can be merged now.

**Comment by BenBE** [comment]
> @BenBE this can be merged now.

Please combine/squash the various commits as mentioned/remarked upon earlier.

Furthermore, please check the spelling in the commit messages, as I think, there are some typos in there.

TIA.

**Comment by pheiduck** [comment]
@BenBE done.

**Comment by BenBE** [comment]
Hm, looks like most of the Ubuntu build tasks are still carried out on Ubuntu 20.04 LTS (focal) instead of Ubuntu 22.04 LTS (Jammy).

**Comment by pheiduck** [comment]
> Hm, looks like most of the Ubuntu build tasks are still carried out on Ubuntu 20.04 LTS (focal) instead of Ubuntu 22.04 LTS (Jammy).

I can apply this to the other build jobs as well.

**Comment by BenBE** [comment]
This seems to break things:
https://github.com/htop-dev/htop/actions/runs/2302402324

**Comment by pheiduck** [comment]
> This seems to break things: https://github.com/htop-dev/htop/actions/runs/2302402324

I thought
i will revert it back  or we use the latest tag for all?

**Comment by pheiduck** [comment]
@BenBE I added gcc-11 as well, what do you think?

**Pull request closed without merging**